### ISSUE TITLE
Add Support for "AND" Find Option

### ISSUE THREAD

**Issue by jesster2k10** [typeorm/typeorm#4360]
**Issue type:**

[ ] question
[ ] bug report
[x] feature request
[ ] documentation issue

**Database system/driver:**
N/A
[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**
N/A
[ ] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

Currently, the only way to do an "AND" operand on a single field is using the query builder like so

```ts
const results = await getRepository(User)
    .createQueryBuilder('user')
    .where('user.id = :id', { id: 1 })
    .andWhere('user.id > :min', { min: 0 })
```

Which when doing dynamic queries can get extremely cumbersome. A much better alternative to this would be introducing a way to do this on the `find` method in the `BaseEntity` through a new FindOperator, or similar, called "And"

So it would look something similar to this:

```ts
const results = await User.find({
    where: {
        id: And(Is(1), GreaterThan(0))
    }
});
```

and produce an sql query similar to 

```sql
SELECT * from "users" "Users" WHERE "Users"."id" = $1 AND "Users"."id" > $2
```

To my knowledge, there's nothing like this implemented already. The closest thing is the ability to use an array opposed to an object in the `where` property which would produce:

```ts
const results = await User.find({
    where:  [
        { id: Is(1) },
        { id: GreaterThan(0) }
    ]
});
```

```sql
SELECT * from "users" "Users" WHERE "Users"."id" = $1 OR "Users"."id" > $2
```

I believe this wouldn't be too hard to implement. Ill see what I can do myself and make a PR.

**Comment by rugglcon** [comment]
@jesster2k10 made any progress on this? i'd like to help if you want, otherwise I could also get started on this.

**Comment by jesster2k10** [comment]
Yeah, I was trying something but I got kind of frustrated waiting forever
for my changes to be compiled locally with the way the test suite was set
up so I eventually just used the Query Builder to save me the hastle but I
think it’s a feature worth adding. If you’ve any progress of some sort I’d
love to see it.

I can push some of the changes I made to a forked repo but they’re far from
working

**Comment by nigelvon** [comment]
Currently, I resolved it like this
```
where: {
  createdAt: Raw(alias => `${alias} >= "2019-07-15 00:00:00"
    AND ${alias} < "2019-07-16 00:00:00"`),
}
```

**Comment by alenap93** [comment]
@jesster2k10  And operator has been implemented with the #9489. I think that you can close this issue...

**Comment by danielR2001** [comment]
@alenap93 what you mentioned is AND on a single field, what is missing is AND on a range of fields.
possible syntax:
```
where: And({
  id: LessThan(3),
}, {
  id: MoreThan(1),
})
```

**Comment by alenap93** [comment]
@danielR2001 i think that the correct way is:
```
where: {
  id: LessThan(3),
  firstName: "John",
}
```

**Comment by vitage** [comment]
Hello, what about this real world case:

```
const where = 
    And({
            status: "progress"
        }, 
        Or({ 
            title: Like("%TEST%")
        },{
            description: Like("%TEST%")
        })
    });
```
